src/Microsoft.OData.Core/UriParser/ODataQueryOptionParser.cs

```diff
@@ -326,6 +331,31 @@ public string ParseDeltaToken()
             string deltaTokenQuery;
             return this.TryGetQueryOption(UriQueryConstants.DeltaTokenQueryOption, out deltaTokenQuery) ? deltaTokenQuery : null;
         }
+
+        /// <summary>
+        /// Parses a compute clause on the given full Uri, binding
+        /// the text into semantic nodes using the constructed mode.
+        /// </summary>
+        /// <returns>A <see cref="ComputeClause"/> representing the computed properties.</returns>
+        public ComputeClause ParseCompute()
+        {
+            if (this.computeClause != null)
+            {
+                return this.computeClause;
+            }
+
+            string computeQuery;
+
+            if (!this.TryGetQueryOption(UriQueryConstants.ComputeQueryOption, out computeQuery)
+                || string.IsNullOrEmpty(computeQuery)
+                || this.targetEdmType == null)

```

Comment:
there are two test cases that address returning null, i believe. both seem to be based on the assertion that all query options must be able to handle null scenarios.

---
In reply to: [123615935](https://github.com/OData/odata.net/pull/827#discussion_r123615935) [](ancestors = 123615935)